cmd/egress-cni-plugin/egressContext.go

```diff
@@ -0,0 +1,302 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/plugins/pkg/ip"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/vishvananda/netlink"
+
+	"github.com/aws/amazon-vpc-cni-k8s/cmd/egress-cni-plugin/snat"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/hostipamwrapper"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/iptableswrapper"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/netlinkwrapper"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/nswrapper"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/procsyswrapper"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/cniutils"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/logger"
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/vethwrapper"
+)
+
+const (
+	ipv4MulticastRange = "224.0.0.0/4"
+)
+
+// EgressContext includes all info to run container ADD/DEL action
+type EgressContext struct {
+	Procsys       procsyswrapper.ProcSys
+	Ipam          hostipamwrapper.HostIpam
+	Link          netlinkwrapper.NetLink
+	Ns            nswrapper.NS
+	NsPath        string
+	ArgsIfName    string
+	Veth          vethwrapper.Veth
+	IpTablesIface iptableswrapper.IPTablesIface
+	IptCreator    func(iptables.Protocol) (iptableswrapper.IPTablesIface, error)
+
+	NetConf   *NetConf
+	Result    *current.Result
+	TmpResult *current.Result
+	Log       logger.Logger
+
+	Mtu     int
+	Chain   string
+	Comment string
```

Comment:
nit: can we add a comment here explaining that this is used for iptables rules? And maybe rename this to `SnatChain` and `SnatComment`?

Reply:
Renamed and add comments